bob/src/hw_metrics_collector.rs

```diff
@@ -360,62 +333,78 @@ impl DiskStatCollector {
         DiskStatCollector {
             procfs_avl: true,
             disk_metric_data,
-            time: TimeContainer::new(),
-            reads: DiffContainer::new(),
-            writes: DiffContainer::new(),
-            read_time: DiffContainer::new(),
-            write_time: DiffContainer::new(),
+            upd_timestamp: Instant::now(),
         }
     }
 
+    fn parse_stat_line(parts: Vec<&str>) -> Result<DiskStats, CommandError> {
+        let mut new_ds = DiskStats {
+            reads: 0.,
+            writes: 0.,
+            write_time: 0.,
+            read_time: 0.,
+            extended: parts.len() >= 11,
+        };
+        if let (Ok(r_ios), Ok(w_ios)) = (
+            // successfull reads count is in 3rd column
+            parts[3].parse::<f64>(),
+            // successfull writes count is in 7th column
+            parts[7].parse::<f64>(),
+        ) {
+            new_ds.reads = r_ios;
+            new_ds.writes = w_ios;  
+            if new_ds.extended {
+                if let (Ok(w_ticks), Ok(r_ticks)) = (
+                    // time spend on write is in 10th column
+                    parts[10].parse::<f64>(),
+                    // time spend on read is in 6th column
+                    parts[6].parse::<f64>(),
+                ) {
+                    new_ds.read_time = r_ticks;
+                    new_ds.write_time = w_ticks;
+                } else {
+                    let msg = format!("Can't parse {} values to float", DISK_STAT_FILE);
+                    return Err(CommandError::Primary(msg));
+                }
+            }
+        } else {
+            let msg = format!("Can't parse {} values to float", DISK_STAT_FILE);
+            return Err(CommandError::Primary(msg));
+        }
+        Ok(new_ds)
+    }
+
     fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
         let diskstats = self.diskstats()?;
-        if let Some(elapsed) = self.time.tick() {
+        let now = Instant::now();
+        let elapsed = now.duration_since(self.upd_timestamp);
+        if elapsed > DIFFCONTAINER_THRESHOLD {
+            self.upd_timestamp = now;
+            let elapsed = elapsed.as_millis() as f64 / 1000.;
```

Comment:
There is a function for that: `as_secs_f64` (https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.as_secs_f64)